### ISSUE TITLE
Accuracy on LFW

### ISSUE THREAD

**Issue by yao5461** [clcarwin/sphereface_pytorch#6]
Hi clcarwin, 
I run the code: python train.py, with default hyper parameters.
and evaluate on lfw: python lfw_eval.py.
The accuracy is:  LFWACC=0.9907 std=0.0033 thd=0.3250.
For your model, the accuracy is: LFWACC=0.9922 std=0.0054 thd=0.3050.
Is there any advice? What else should I do?
Thanks!

**Comment by clcarwin** [comment]
You can try to set ```phiflag``` to ```False``` and fine-tune more epoch on your model.

**Comment by HaoLiuHust** [comment]
@clcarwin I have tried that, after 5 epoch, loss came to nan, and acc will be 50%

**Comment by bixiang** [comment]
@yao5461  Can you give me your email? I have some questions about sphereface_pytorch. Thanks very much!

**Comment by YihangLou** [comment]
@HaoLiuHust  I meet the same problem, have you solved it? Thanks, if you can share a solution for working around.

**Comment by HaoLiuHust** [comment]
@YihangLou change the learning rate to a smaller number, however, I can not get the acc reported here

**Comment by YihangLou** [comment]
thank you i solve it in the same way. the performce is still under the expectation

发自我的 iPhone

在 2018年3月2日，11:02，Liu, Hao <notifications@github.com<mailto:notifications@github.com>> 写道：


@YihangLou<https://github.com/yihanglou> change the learning rate to a smaller number, however, I can not get the acc reported here

**Comment by Gerkam** [comment]
I train on washed casia-web-face(~450k) and test on lfw with accuracy-61%. All params was default. What can be problem?

**Comment by vicdu** [comment]
@yao5461 Have u got a good work when set phiflag to False?

**Comment by Gerkam** [comment]
@yao5461 Did you train on original or cleaned Casisa-WebFace?

**Comment by zeakey** [comment]
@yao5461 Hi Yao, could you please provide the `lfw.zip` and `lfw_landmark.txt` that are used in the evaluation? I cannot reproduce the evaluation results even with the author released models, which is claimed to be 0.9922.

**Comment by timho102003** [comment]
@Gerkam Can u provide a valid link to download cleaned Casia-Webface washed list?
I've searching it for a while and can't find a valid link.
Thanks!!!!!!

**Comment by Gerkam** [comment]
@timho102003 See https://groups.google.com/forum/#!topic/cmu-openface/Xue_D4_mxDQ and https://github.com/cmusatyalab/openface/issues/119

**Comment by timho102003** [comment]
@Gerkam I've found these websites too, however the links are all dead. Do u have other valid download link? 
Thanks!!!!!

**Comment by Gerkam** [comment]
@timho102003 The washed list can be downloaded from http://pan.baidu.com/s/1kUdRRJT with password 3zbb . For fast download find manuals how can you do it from baidu

**Comment by timho102003** [comment]
@Gerkam when entering the password, baidu shows that the content is illegal and cannot be downloaded. I don't really understand ur last sentence. Does that mean I should find a way to download such data from baidu??
BTW Thanks for ur prompt response!

**Comment by Gerkam** [comment]
@timho102003 After enter password I see all files. So, maybe baidu block request from your country or another reason. In previous post I mean, if you want download with good speed you must see how can you do it. By default speed will be slow.

**Comment by RyanCV** [comment]
@clcarwin @yao5461 Is there an error in the lfw_eval code?
 https://github.com/clcarwin/sphereface_pytorch/blob/e2f11f0b0f394cf75b9f65ce8d353e4390b0948c/lfw_eval.py#L114 

in the original paper it said the original and flipped image features are concatenated together, and then to calculate the cosine similarity. 

As shown here: https://github.com/wy1iu/sphereface/blob/a848797f1b1da00ef756ea73b38b4926cc3563cd/test/code/evaluation.m#L112.

**Comment by zorrocai** [comment]
@RyanCV I have the same question. Do you think it should be changed to: f1,f2 = [f[0],f[1]], [f[2],f[3]]?
